### PR TITLE
Fix msgstream close block.

### PR DESCRIPTION
**Pull request by sunby**
issue: #17513
Signed-off-by: sunby <bingyi.sun@zilliz.com>

### CODE REVIEW

**Comment by xiaofan-luan** [comment]
good catch, this is very hard to reproduce I guess

File: internal/mq/msgstream/mq_msgstream.go [hunk 1]
```diff
@@ -185,11 +183,12 @@ func (ms *mqMsgStream) Start() {
 }
 
 func (ms *mqMsgStream) Close() {
-
 	ms.streamCancel()
-	ms.readerLock.Lock()
+	if len(ms.receiveBuf) > 0 {
```
Comment (czs007) [inline internal/mq/msgstream/mq_msgstream.go:187]: better to write like this:
select {
      case _= <- ms.receiveBuf:
     default:
}

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**